pkg/proxy/k8s.go

```diff
@@ -72,15 +74,25 @@ func (kc *KubernetesClient) InitClient(address, token string, gateway *model.Gat
 		address = ReplaceURLHostAndPort(originUrl, "127.0.0.1", proxyAddr.Port)
 	}
 
-	kubeConfig := kc.GetKubeConfig(address, token)
+	kubeConf := &rest.Config{
+		Host:        address,
+		BearerToken: token,
+	}
+	kubeConf.Insecure = true
+
+	if !srvconn.IsValidK8sUserToken(kubeConf) {
+		return srvconn.ErrValidToken
+	}
+
+	kubeConfigYAML := kc.GetKubeConfig(address, token)
 
 	tmpFile, err := os.CreateTemp("", "kubeconfig-*.yaml")
 	if err != nil {
 		return fmt.Errorf("error creating temp file: %w", err)
 	}
 	defer tmpFile.Close()
 
-	if _, err := tmpFile.Write([]byte(kubeConfig)); err != nil {
+	if _, err := tmpFile.Write([]byte(kubeConfigYAML)); err != nil {
 		return fmt.Errorf("error writing to temp file: %w", err)
 	}
 	kc.configName = tmpFile.Name()
```

Comment:
I noticed that there is an error in the function InitializeKubernetesClient(). The provided code snippet seems to refer to initializing clients using various configurations such as Kubernetes Config and RESTConfig. However, it does not contain the implementation of those configurations.

For example:
- `kc.GetKubeConfig()`: Returns a kube config object which contains authentication details needed for accessing the specified host and port.
- `GetKubeConfig()` returns the value assigned to cfg variable when called with parameters like "host" and "token".

Therefore, I'm unable to analyze what errors are present or suggest optimizations in the given piece of code without implementing the configurations first.